public_html/wp-content/plugins/wordcamp-organizer-survey/includes/debrief-survey/page.php

```diff
@@ -0,0 +1,341 @@
+	<!-- wp:jetpack/field-option-checkbox {"label":"Doable (less human and financial resources)"} /-->
+	
+	<!-- wp:jetpack/field-option-checkbox {"label":"Replicable (by other communities in different parts of the world)"} /-->
+	
+	<!-- wp:jetpack/field-option-checkbox {"label":"Scalable (can be changed in size or scale)"} /-->
+	
+	<!-- wp:jetpack/field-option-checkbox {"label":"Desirable (seen as useful, important or necessary by the community)"} /-->
+	<!-- /wp:jetpack/field-checkbox-multiple --></div>
+	<!-- /wp:group -->
+	
+	<!-- wp:group {"layout":{"type":"constrained"}} -->
+	<div class="wp-block-group"><!-- wp:jetpack/field-radio {"label":"Did you have enough support from WordCamp Central?","required":true,"requiredText":"(required)"} -->
+	<!-- wp:jetpack/field-option-radio {"label":"Yes."} /-->
+	
+	<!-- wp:jetpack/field-option-radio {"label":"No."} /-->
+	<!-- /wp:jetpack/field-radio --></div>
+	<!-- /wp:group -->
+	
+	<!-- wp:group {"layout":{"type":"constrained"}} -->
+	<div class="wp-block-group"><!-- wp:jetpack/field-textarea {"label":"If there were things that made planning this event harder or easier, let us know!","requiredText":"(required)"} /--></div>
+	<!-- /wp:group -->
+	
+	<!-- wp:group {"layout":{"type":"constrained"}} -->
+	<div class="wp-block-group"><!-- wp:jetpack/field-text {"label":"Have you written a recap post of the event yet? If so, please provide the URL so we can reblog it on WordCamp Central website.","required":true,"requiredText":"(required)"} /--></div>
+	<!-- /wp:group -->
+	
+	<!-- wp:group {"layout":{"type":"constrained"}} -->
+	<div class="wp-block-group"><!-- wp:jetpack/field-textarea {"label":"Is there anything else we should know about how your event went?","requiredText":"(required)"} /--></div>
+	<!-- /wp:group --></div>
+	<!-- /wp:group -->
+	
+	<!-- wp:paragraph -->
+	<p>Thank you so much for sharing your feedback and providing an excellent experience to your community, we are very grateful for your contribution to the WordPress open source project!</p>
+	<!-- /wp:paragraph -->
+	
+	<!-- wp:jetpack/button {"element":"button","text":"Send Feedback","lock":{"remove":true}} /--></div>
+	<!-- /wp:jetpack/contact-form -->
+EOT;
+}
+
+/**
+ * Turns on the page by changing its status to 'publish'.
+ *
+ * @return int|WP_Error
+ */
+function publish_survey_page() {
+	return wp_update_post( array(
+		'ID'            => get_option( SURVEY_PAGE_ID ),
+		'post_status'   => 'publish',
+	) );
+}
+
+/**
+ * Generate a link to the front end feedback UI for a particular session.
+ *
+ * @return string|bool
+ */
+function get_survey_page_url() {
+	return get_permalink( get_option( SURVEY_PAGE_ID ) );
+}
+
+
+/**
+ * Create the Survey page, save ID into an option.
+ */
+function add_page() {
+	$page_id = get_option( SURVEY_PAGE_ID );
+
+	if ( $page_id ) {
+		return;
+	}
+
+	$page_id = wp_insert_post( array(
```

Comment:
🤔 I wonder if it's better to create a single page on Central, rather than a new page on each camp site? There'll only be 1 respondent per camp, so it seems it might be overkill to do it on every site.

It could remove the need for `admin-page.php`, code to create/publish/delete/protect the page, etc.

I don't feel strongly, though. What do you think?

Reply:
> I wonder if it's better to create a single page on Central, rather than a new page on each camp site? There'll only be 1 respondent per camp, so it seems it might be overkill to do it on every site.

If we only create a page on the central site, then we can't have the mechanism to set this page as a draft to prevent spam. 

We currently have an anti-spam mechanism where, two weeks after an event site concludes, this published page is automatically deleted to prevent organizers from accessing it. 

If we were to create a page on the central site, the timing of changing the page status from draft to publish (add the page to camptix email) triggered by one site would overlap with the anti-spam mechanism's change from publish to draft triggered by another site. 

That said, given that only one person, the lead organizer, is filling it out the survey for each event, the chances of spam, while still possible, are considerably reduced.

> It could remove the need for admin-page.php, code to create/publish/delete/protect the page, etc.

I understand we can remove the need for code to create/publish/delete/protect the page if moving the page to central, but I'm a bit not sure how this can also affect the need for admin-page.php.

Reply:
> then we can't have the mechanism to set this page as a draft to prevent spam.

🤔 That's a good point. We could manually create the page on Central, and leave it published all the time. We then might be able to require the organizer to log in to submit an entry, and we could automatically populate the "which wordcamp..." field based on that. We could show them an error if their camp was over more than 2 weeks ago, or maybe we wouldn't have to worry about that because we know only the organizer can fill it in.

Reply:
>  how this can also affect the need for admin-page.php.

I was thinking that it wouldn't be needed, since admins could use a workflow like this:

1. Visit https://central.wordcamp.org/wp-admin/edit.php?post_type=feedback
2. change `All sources` to the specific form
3. click `Filter` or `Export`

Reply:
🤔 Rather than requiring them to log in, we could also just put a token in the URL that we send them, and use that to verify that they're an organizer. That might avoid some folks skipping it because of the extra work of logging in.

It may also avoid some edge cases where there's extra hassles. For example, we store the email address of the lead organizer, but not their w.org username. So if they log in but their w.org account email doesn't match the email we sent the request to, then they might have to update the email to match.

For the token, we might be able to just create a secret key in the config, and then create a hash of their email address and the key; something like `hash_hmac( 'sha1', $email, ORGANIZER_SURVEY_ACCESS_TOKEN_KEY )`. That way we wouldn't have to store any additional metadata in the database. If having the value in the db is needed for other parts of the code, though, then that's not a problem either. In that case we could probably just use `wp_generate_password()` to create the token.

Reply:
Thanks for the suggestion 👍 👍 👍 
I've removed the admin page and moved the survey page to central.

And after talking a bit with Steve earlier today, we figure it seems we can get more things on the central site if including the `wordcamp_id` in the URL - capable of answering [as many questions as possible](https://github.com/WordPress/wordcamp.org/pull/1038#discussion_r1329418934) with the event site data retrieved from it. 
So it would be `hash_hmac( 'sha1', $wordcamp_id, ORGANIZER_SURVEY_ACCESS_TOKEN_KEY )`.

The changes starts from this commit [7354b3a](https://github.com/WordPress/wordcamp.org/pull/1038/commits/7354b3a7d7e13dc43a3a2b7f65cf34810e05ef21)